develop/blockchains/custom-blockchains/benchmarking.md

```diff
@@ -1,4 +1,183 @@
 ---
-title: Benchmarking
-description: TODO
----
\ No newline at end of file
+title: Benchmark Testing
+description: Learn how to use FRAME's benchmarking framework to benchmark your custom pallets and provide correct weights for on-chain computation and execution of extrinsics.
+---
+
+# Benchmark Testing
+
+## Introduction
+
+Benchmark testing is a critical component of developing efficient and secure blockchain runtimes. In the Polkadot ecosystem, accurately benchmark testing your custom pallets ensures that each extrinsic has a precise [weight](../../../polkadot-protocol/glossary.md#weight), representing its computational and storage demands. This process is vital for maintaining the blockchain's performance and preventing potential vulnerabilities, such as Denial of Service (DoS) attacks.
```

Comment:
Please check all of the links and update to use absolute links. Also please check to make sure `{target=\_blank}` is there